src/main/java/seedu/taassist/logic/parser/DeletecCommandParser.java

```diff
@@ -21,10 +23,22 @@ public class DeletecCommandParser implements Parser<DeletecCommand> {
     public DeletecCommand parse(String args) throws ParseException {
         try {
             ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_MODULE_CLASS);
-            ModuleClass moduleClass = ParserUtil.parseModuleClass(argMultimap.getValue(PREFIX_MODULE_CLASS).get());
-            return new DeletecCommand(moduleClass);
+            if (!isPrefixPresentAndNotEmpty(argMultimap, PREFIX_MODULE_CLASS)) {
+                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeletecCommand.MESSAGE_USAGE));
+            }
+
+            List<String> moduleClassStrings = argMultimap.getAllValues(PREFIX_MODULE_CLASS);
+            Set<ModuleClass> moduleClasses = ParserUtil.parseModuleClasses(moduleClassStrings);
+            return new DeletecCommand(moduleClasses);
         } catch (ParseException pe) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE), pe);
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeletecCommand.MESSAGE_USAGE), pe);
         }
     }
+
+    /**
+     * Returns true if the prefix for the command is present and non-empty.
+     */
+    private static boolean isPrefixPresentAndNotEmpty(ArgumentMultimap argumentMultimap, Prefix prefix) {
```

Comment:
This function does not sound like it belongs here. 

Currently there are similar methods in other `*Parser` classes too. Wouldn't it be better to move them into a util class?

Reply:
Good idea, I shifted it to `ParserUtil`.